gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/FieldSchema.java

```diff
@@ -0,0 +1,287 @@
+/*
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.bigquery;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.api.services.bigquery.model.TableFieldSchema;
+import com.google.common.base.Function;
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Google Bigquery schema for a Table field. Several data types and modes are supported.
+ */
+public class FieldSchema implements Serializable {
+
+  static final Function<TableFieldSchema, FieldSchema> FROM_PB_FUNCTION =
+      new Function<TableFieldSchema, FieldSchema>() {
+        @Override
+        public FieldSchema apply(TableFieldSchema pb) {
+          return FieldSchema.fromPb(pb);
+        }
+      };
+  static final Function<FieldSchema, TableFieldSchema> TO_PB_FUNCTION =
+      new Function<FieldSchema, TableFieldSchema>() {
+        @Override
+        public TableFieldSchema apply(FieldSchema fieldSchema) {
+          return fieldSchema.toPb();
+        }
+      };
+
+  private static final long serialVersionUID = -8154262932305199256L;
+
+  /**
+   * Data Types for a BigQuery Table field.
+   *
+   * @see <a href="https://cloud.google.com/bigquery/preparing-data-for-bigquery#datatypes">
+   *     Data Types</a>
+   */
+  public enum Type {
```

Comment:
Another option that works nicer with RECORDS (and fields) is to make that a class with private constructor.
The class could have an an inner enum `Value` with the same set  of values.
Class has a factory method for each type `string()`, `integer()`, ... record(FieldSchema...)`.